pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -768,6 +814,85 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    pub(crate) async fn delete_all(self: &Arc<Self>) -> anyhow::Result<()> {
+        let mut receiver = {
+            let mut locked = self.upload_queue.lock().unwrap();
+            let stopped = match &mut *locked {
+                UploadQueue::Uninitialized => {
+                    anyhow::bail!("upload queue is not stopped but initialized")
+                }
+                UploadQueue::Initialized(_) => {
+                    anyhow::bail!("upload queue is not stopped but initialized")
+                }
+                UploadQueue::Stopped(stopped) => stopped,
+            };
```

Comment:
If you would like to, I think this is now in few places, we could refactor it to a method, or even `TryFrom`.

There's a typo in the first message (initialized => uninitialized).

Reply:
Which try from you were thinking about? Stopped::try_from(upload_queue)? 

I'm thinking of making stopped_mut akin to initialized_mut we have

Reply:
`stopped_mut` sounds good. Not sure what I was thinking of, probably just a try_from conversion in place of the `stopped_mut` conversion, but that should work just as well and be require less named lifetimes.

Reply:
Done